location_extractor/clients.py

```diff
@@ -0,0 +1,252 @@
+
+from location_extractor import src_dir
+from location_extractor.helpers import remove_accents
+import csv
+
+
+@dataclass(frozen=True, order=True)
+class LocationDTO:
+    locale_code: str
+    continent_code: str
+    continent_name: str
+    country_iso_code: str
+    country_name: str
+    subdivision_name: str
+    city_name: str
+    is_in_european_union: bool
+
+
+class DBClient:
+    def __init__(self) -> None:
+        self.dbpath = os.path.join(src_dir, "data", "data.db")
+
+        self.index_columns = [
+            'continent_name',
+            'country_iso_code',
+            'country_name',
+            'subdivision_name',
+            'city_name',
+        ]
+
+        with self.connect() as conn:
+            table_exists = conn.execute("""
+                SELECT name FROM sqlite_master WHERE type='table' AND name='locations';
+            """).fetchone()
+
+            if not table_exists:
+                conn.execute('''
+                    CREATE TABLE locations
+                    (
+                        locale_code text,
+                        continent_code text,
+                        continent_name text,
+                        country_iso_code text,
+                        country_name text,
+                        subdivision_name text,
+                        city_name text,
+                        is_in_european_union boolean,
+                        locale_code_lowercase text,
+                        continent_code_lowercase text,
+                        continent_name_lowercase text,
+                        country_iso_code_lowercase text,
+                        country_name_lowercase text,
+                        subdivision_name_lowercase text,
+                        city_name_lowercase text
+                    )
+                    ''')
+
+                for col in self.index_columns:
+                    conn.execute(f'''
+                        CREATE INDEX {col} 
+                        ON locations({col}_lowercase);
+                    ''')
+
+                locations_path = os.path.join(
+                    src_dir, "data", "GeoLite2-City-CSV_20200303", "GeoLite2-City-Locations-en-processed.csv")
+
+                locations_data = []
+                with open(locations_path, 'r') as locations_file:
+                    reader = csv.reader(locations_file, delimiter=',')
+                    columns = next(reader)
+                    for i in range(len(columns) - 1):
+                        columns.append(f'{col}_lowercase')
+
+                    for row in reader:
+                        locale_code = row[0] or ''
+                        continent_code = row[1] or ''
+                        continent_name = row[2] or ''
+                        country_iso_code = row[3] or ''
+                        country_name = row[4] or ''
+                        subdivision_name = row[5] or ''
+                        city_name = row[6] or ''
```

Comment:
too avoid this quite long chunk of code we can refactor appendig to `locations` in following way:

```suggestion
                        str_values = [value or '' for value in row[:7]]
```

then ;let's use iterable unpacking and eventually generator expressions to create tuple that is going to be appended to `locations`:

```python
locations.append((
    *str_values,
    is_in_european_union,
    *(value.lower() for value in str_values),
))
```